src/event.rs

```diff
@@ -225,107 +139,61 @@ pub enum WindowEvent<'a> {
     ///
     /// When the user drops multiple files at once, this event will be emitted for each file
     /// separately.
-    DroppedFile(PathBuf),
+    FileDropped(PathBuf),
 
     /// A file is being hovered over the window.
     ///
     /// When the user hovers multiple files at once, this event will be emitted for each file
     /// separately.
-    HoveredFile(PathBuf),
+    FileHovered(PathBuf),
 
     /// A file was hovered, but has exited the window.
     ///
     /// There will be a single `HoveredFileCancelled` event triggered even if multiple files were
     /// hovered.
-    HoveredFileCancelled,
+    FileHoverCancelled,
+
+    /// The window gained focus.
+    FocusGained,
+
+    /// The window lost focus.
+    FocusLost,
 
     /// The window received a unicode character.
-    ReceivedCharacter(char),
+    CharReceived(char),
 
-    /// The window gained or lost focus.
-    ///
-    /// The parameter is true if the window has gained focus, and false if it has lost focus.
-    Focused(bool),
-
-    /// An event from the keyboard has been received.
-    KeyboardInput {
-        device_id: DeviceId,
-        input: KeyboardInput,
-        /// If `true`, the event was generated synthetically by winit
-        /// in one of the following circumstances:
-        ///
-        /// * Synthetic key press events are generated for all keys pressed
-        ///   when a window gains focus. Likewise, synthetic key release events
-        ///   are generated for all keys pressed when a window goes out of focus.
-        ///   ***Currently, this is only functional on X11 and Windows***
-        ///
-        /// Otherwise, this value is always `false`.
-        is_synthetic: bool,
-    },
+    KeyPress(KeyPress),
 
     /// The keyboard modifiers have changed.
     ///
+    /// This is tracked internally to avoid tracking errors arising from modifier key state changes when events from
+    /// this device are not being delivered to the application, e.g. due to keyboard focus being elsewhere.
+    ///
     /// Platform-specific behavior:
     /// - **Web**: This API is currently unimplemented on the web. This isn't by design - it's an
     ///   issue, and it should get fixed - but it's the current state of the API.
     ModifiersChanged(ModifiersState),
 
-    /// The cursor has moved on the window.
-    CursorMoved {
-        device_id: DeviceId,
+    PointerCreated(PointerId),
+    PointerForce(PointerId, Force),
+    PointerMoved(PointerId, PhysicalPosition<f64>),
+    PointerPress(PointerId, PointerPress),
+    PointerEntered(PointerId),
+    PointerLeft(PointerId),
+    PointerDestroyed(PointerId),
```

Comment:
What's the idea behind `PointerCreated` and `PointerDestroyed`?

Reply:
Those are mainly useful for touch events, since placing a new finger in contact with the screen creates a new pointer for that finger.

Reply:
I feel like separating mouse pointer and touch events is a good idea and could make such things more clear?

Reply:
Unifying mouse, touch, and pen events into a single event class was actually an explicit goal for this rework, since that allows applications that do the naive thing and only handle one pointer type to get basic support for other pointer types. Janky touch support in applications designed for mouse input is better than no touch support.

Reply:
But at this point you just want to cut proper touch/tablet at all? I feel like you can add flag to translate pen/touch into mouse events in the end for such clients. What if I have 2 mouse cursors, how should I handle it, since it could be 2 touch presses?

>Janky touch support in applications designed for mouse input is better than no touch support

If the goal is to translate pen/touch, you can add a flag to do so, I guess. on Wayland compositor can do that for you for example, however it requires to not bind touch at all.

Reply:
> I feel like you can add flag to translate pen/touch into mouse events in the end for such clients.

There's probably some benefit to making multipointer support opt-in, but there's no need to split mouse events and rich pointer events into separate events when doing that. Doing that would needlessly add redundancy to the API.

> What if I have 2 mouse cursors, how should I handle it, since it could be 2 touch presses?

Touch events can still be distinguished from mouse events by inspecting the `PointerId` type.